src/Core/Domain/Category/Command/AbstractRootCategoryCommand.php

```diff
@@ -0,0 +1,363 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Category\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryConstraintException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Class AbstractAddCategoryCommand defines common command for both simple and root category creation.
+ */
+abstract class AbstractRootCategoryCommand
+{
+    /**
+     * @var string[]
+     */
+    private $name;
```

Comment:
Why is it `$name` instead of `$names` if this is an array ?

Reply:
well, i had idea of naming it as `$names` but is it really a correct way of doing this, isnt it confusing?

does this mean that Category can have many different names? or does it mean that category has single name, but in different languages? :thinking: see my point? either way, im not against `$names` if that makes more sense for you. :)